File: Java_Codes/9. Multithreading/RunnableExample.java
//Downloading files or processing network requests concurrently.
class MyRunnable implements Runnable {
    @Override
    public void run() {
        System.out.println("Runnable thread is running: " + Thread.currentThread().getName());
    }
}

public class RunnableExample {
    public static void executeRunnableExample() {
        Thread thread = new Thread(new MyRunnable());
        thread.start();
        Thread thread1 = new Thread(new MyRunnable());
        thread1.start();
    }

    public static void main(String[] args) {
        executeRunnableExample();
    }
}

File: Python_Codes/9. Multithreading/RunnableExample.py
import threading

class MyRunnable:
    def run(self):
        print(f"Runnable thread is running: {threading.current_thread().name}")

def execute_runnable_example():
    runnable = MyRunnable()
    
    # Create and start two threads
    thread1 = threading.Thread(target=runnable.run)
    thread2 = threading.Thread(target=runnable.run)
    
    thread1.start()
    thread2.start()

if __name__ == "__main__":
    execute_runnable_example()
